lib/Day09Year2023/Predictor.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Lib.Day09Year2023
+{
+    public class Predictor
+    {
+        public List<int> GetPredicates(List<List<int>> report)
+        {
+            List<int> predicats = [];
+            foreach (List<int> history in report)
+            {
+                List<List<int>> offset = BuildOffset([history]);
+                List<List<int>> nextValue = NextValue(offset);
+                predicats.Add(nextValue[0][^1]);
+            }
+
+            return predicats;
+        }
+        public List<List<int>> BuildOffset(List<List<int>> history) 
+        {       
+            bool zero = history[^1].All(x => x == 0);
```

Comment:
simplifier comme quand on a fait `Select(int.Parse)` au lieu de `Select(x => int.Parse(x))`

```suggestion
            bool zero = history[^1]
                .All(x => x == 0);
```